### PR TITLE
[TASK] Streamline CI-Setup

### PR DESCRIPTION
**Pull request by benjaminkott**
The goal of this patch is to provide a base for multi env
testing of the composer installer. The base matrix now
also includes composer versions 1 and 2.

The missing LICENSE.txt file was added.
The unused .styleci file was removed.

Git is now configured and will exclude ci related files
from shipped packages, in addition, we ensure that 
relevant files are checked out with LF line endings 
across all platforms.

Editorconfig will help us to ensure consistent line endings 
and indentions across the project.

CGL check is now enabled again for ci builds, we are using a 
standalone package for php-cs-fixer to avoid version conflicts
with shared dependencies of the installer. The header comment
is now also covered through the CGL to ensure it´s available
in all files.

For linting we are now using the overtrue/phplint package, this also
allows easy linting on non-linux systems and provides better ci fitting 
reporting.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -4,22 +4,48 @@ on: [push, pull_request]
...
+          composer info
+
+      - name: Lint
+        run: |
+          composer test:php:lint
+
+      - name: CGL
         run: |
-          find . -name \*.php -exec php -l {} >/dev/null \;
+          php-cs-fixer fix --dry-run
```
Comment (helhum) [inline .github/workflows/ci.yml:51]: can you add a `--verbose` here?

File: .editorconfig [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+max_line_length = 80
+
+# YAML-Files
+[*.{yaml,yml}]
+indent_size = 2
+
+# package.json
+# .travis.yml
+[{package.json,.travis.yml}]
+indent_size = 2
```
Comment (helhum) [inline .editorconfig:35]: indent size 4 for yaml please

File: .editorconfig [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+indent_style = tab
+
+# ReST-Files
+[*.rst]
+indent_size = 3
+max_line_length = 80
+
+# YAML-Files
+[*.{yaml,yml}]
+indent_size = 2
```
Comment (helhum) [inline .editorconfig:30]: indent size 4 for yaml please

File: .editorconfig [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+insert_final_newline = true
+trim_trailing_whitespace = true
+
+# TS/JS-Files
+[*.{ts,js}]
+indent_size = 2
+
+# JSON-Files
+[*.json]
+indent_style = tab
```
Comment (helhum) [inline .editorconfig:21]: indent style space

File: .editorconfig [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+charset = utf-8
+end_of_line = lf
+indent_style = space
+indent_size = 4
+insert_final_newline = true
+trim_trailing_whitespace = true
+
+# TS/JS-Files
+[*.{ts,js}]
+indent_size = 2
```
Comment (helhum) [inline .editorconfig:17]: not sure we have ts/js files at any point in time, but still prefer indent size 4 for all files

File: .editorconfig [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
+# JSON-Files
+[*.json]
+indent_style = tab
+
+# ReST-Files
+[*.rst]
+indent_size = 3
```
Comment (helhum) [inline .editorconfig:25]: size 3 is required for rst? if not, I'd go for 4 as well

File: .editorconfig [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+indent_size = 2
+
+# package.json
+# .travis.yml
+[{package.json,.travis.yml}]
+indent_size = 2
+
+# TypoScript
+[*.{typoscript,tsconfig}]
+indent_size = 2
```
Comment (helhum) [inline .editorconfig:39]: 4

File: .editorconfig [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+
+# XLF-Files
+[*.xlf]
+indent_style = tab
+
+# SQL-Files
+[*.sql]
+indent_style = tab
+indent_size = 2
```
Comment (helhum) [inline .editorconfig:48]: 4

File: composer.json [hunk 9]
```diff
@@ -1,51 +1,60 @@
...
-    "require-dev": {
-        "composer/composer": "1.2.*@dev || 2.0.*@dev"
-    },
-    "extra": {
-        "class": "TYPO3\\CMS\\Composer\\Installer\\Plugin",
-        "branch-alias": {
-            "dev-master": "3.0.x-dev"
-        }
-    }
+	"name": "typo3/cms-composer-installers",
```
Comment (helhum) [inline composer.json:2]: I prefer spaces over tabs in composer.json

File: src/Plugin/Core/IncludeFile/AppDirToken.php [hunk 10]
```diff
@@ -17,6 +13,13 @@
  * The TYPO3 project - inspiring people to share!
  */
 
+namespace TYPO3\CMS\Composer\Plugin\Core\IncludeFile;
+
+/**
+ * This file was taken from the typo3 console plugin package.
+ * (c) Helmut Hummel <info@helhum.io>
+ */
```
Comment (helhum) [inline src/Plugin/Core/IncludeFile/AppDirToken.php:21]: I'm fine to remove this now. plugin package is deprecated anyway and the canonical implementation is here

File: src/Plugin/Core/IncludeFile/BaseDirToken.php [hunk 11]
```diff
@@ -17,6 +13,13 @@
  * The TYPO3 project - inspiring people to share!
  */
 
+namespace TYPO3\CMS\Composer\Plugin\Core\IncludeFile;
+
+/**
+ * This file was taken from the typo3 console plugin package.
+ * (c) Helmut Hummel <info@helhum.io>
+ */
+
```
Comment (helhum) [inline src/Plugin/Core/IncludeFile/BaseDirToken.php:22]: I'm fine to remove this now. plugin package is deprecated anyway and the canonical implementation is here

File: src/Plugin/Core/IncludeFile/TokenInterface.php [hunk 12]
```diff
@@ -17,6 +13,12 @@
  * The TYPO3 project - inspiring people to share!
  */
 
+namespace TYPO3\CMS\Composer\Plugin\Core\IncludeFile;
+
+/**
+ * This file was taken from the typo3 console plugin package.
+ * (c) Helmut Hummel <info@helhum.io>
+ */
```
Comment (helhum) [inline src/Plugin/Core/IncludeFile/TokenInterface.php:21]: I'm fine to remove this now. plugin package is deprecated anyway and the canonical implementation is here

File: src/Plugin/Core/IncludeFile/WebDirToken.php [hunk 13]
```diff
@@ -17,6 +13,13 @@
  * The TYPO3 project - inspiring people to share!
  */
 
+namespace TYPO3\CMS\Composer\Plugin\Core\IncludeFile;
+
+/**
+ * This file was taken from the typo3 console plugin package.
+ * (c) Helmut Hummel <info@helhum.io>
+ */
+
```
Comment (helhum) [inline src/Plugin/Core/IncludeFile/WebDirToken.php:22]: I'm fine to remove this now. plugin package is deprecated anyway and the canonical implementation is here

File: src/Plugin/Core/ScriptDispatcher.php [hunk 14]
```diff
@@ -18,6 +14,13 @@
  * The TYPO3 project - inspiring people to share!
  */
 
+namespace TYPO3\CMS\Composer\Plugin\Core;
+
+/**
+ * This file was taken from the typo3 console plugin package.
+ * (c) Helmut Hummel <info@helhum.io>
+ */
```
Comment (helhum) [inline src/Plugin/Core/ScriptDispatcher.php:22]: I'm fine to remove this now. plugin package is deprecated anyway and the canonical implementation is here

File: src/Plugin/PluginImplementation.php [hunk 15]
```diff
@@ -18,6 +14,13 @@
  * The TYPO3 project - inspiring people to share!
  */
 
+namespace TYPO3\CMS\Composer\Plugin;
+
+/**
+ * This file was taken from the typo3 console plugin package.
+ * (c) Helmut Hummel <info@helhum.io>
+ */
```
Comment (helhum) [inline src/Plugin/PluginImplementation.php:22]: I'm fine to remove this now. plugin package is deprecated anyway and the canonical implementation is here

File: .editorconfig (hunk 5 above)
Comment (benjaminkott) [reply .editorconfig:17]: just used the file from core, will make the adjustments as you noted.

File: .editorconfig (hunk 2 above)
Comment (benjaminkott) [reply .editorconfig:35]: will remove these as travis or node is not used here

File: .editorconfig (hunk 4 above)
Comment (benjaminkott) [reply .editorconfig:21]: will remove as fallbacks to default then

File: .editorconfig (hunk 3 above)
Comment (benjaminkott) [reply .editorconfig:30]: will remove as fallbacks to default then

File: .editorconfig (hunk 6 above)
Comment (benjaminkott) [reply .editorconfig]: rst is recommended for 3, but since we don't have any rst files currently removed the definition

File: .editorconfig (hunk 8 above)
Comment (benjaminkott) [reply .editorconfig]: also removed since we have no sql files

File: composer.json (hunk 9 above)
Comment (benjaminkott) [reply composer.json]: adjusted, to new editor config settings

File: .editorconfig (hunk 7 above)
Comment (benjaminkott) [reply .editorconfig]: removed as we have no typescript and tsconfig

**Comment by benjaminkott** [comment]
will reopen a new pr

**Pull request closed without merging**